tests/unit-tests/gpartition/test_random_partition.py

```diff
@@ -13,11 +13,59 @@
     See the License for the specific language governing permissions and
     limitations under the License.
 """
+import pytest
+from tempfile import TemporaryDirectory
+
 from graphstorm.gpartition import RandomPartitionAlgorithm
 
 from conftest import simple_test_partition
 
-
-def test_create_random_partition(chunked_metadata_dict):
+def test_create_random_partition_basic():
+    chunked_metadata_dict = {
+        "num_nodes_per_type": [10, 20, 1024],
+        "node_type": ["a", "b", "c"],
+    }
     rand_partitioner = RandomPartitionAlgorithm(chunked_metadata_dict)
-    simple_test_partition(rand_partitioner, "random", chunked_metadata_dict)
+    simple_test_partition(
+        rand_partitioner,
+        "random",
+        num_parts=4,
+        chunked_metadata_dict=chunked_metadata_dict
+    )
+
+
+@pytest.mark.parametrize("num_nodes_parts", [1, 2, 8, 256, 512])
+def test_random_partition_equal_nodes_parts(num_nodes_parts: int):
+    """Test random partition when num_nodes==num_partitions"""
+    chunked_meta_dict = {
+        "num_nodes_per_type": [num_nodes_parts],
+        "node_type": ["a"],
+    }
+
+    rand_partitioner = RandomPartitionAlgorithm(chunked_meta_dict)
+    # Repeat assignment ten times to get random results
+    for _ in range(10):
+        simple_test_partition(
```

Comment:
why we what to assign ten times?

Reply:
Explained in the comment above, we just want to be more certain that the specific random assignment just happened to have one node in each partition, so we repeat the process ten times to increase confidence.